File: frontend/src/api/eval.ts
import { request } from "./http";

export type EvalRun = {
  id: string;
  kind: string;
  status: string;
  input_text: string;
  metrics?: Record<string, unknown> | null;
  detail?: Record<string, unknown> | null;
  created_at: string;
};

export const evalApi = {
  evalRuns: () => request<EvalRun[]>("/api/admin/eval"),
  evalQuestions: (jobText: string) =>
    request<EvalRun>("/api/admin/eval/questions", { method: "POST", body: JSON.stringify({ job_text: jobText }) }),
  evalScores: (interviewId?: string) =>
    request<EvalRun>("/api/admin/eval/scores", {
      method: "POST",
      body: JSON.stringify({ interview_id: interviewId, repeats: 5 }),
    }),
};

File: backend/app/schemas/eval.py
"""Request and response shapes for admin quality-eval runs."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class EvalRunOut(BaseModel):
    id: str
    kind: str
    status: str
    input_text: str
    metrics: dict[str, Any] | None = None
    detail: dict[str, Any] | None = None
    created_at: datetime


class EvalQuestionIn(BaseModel):
    job_text: str


class EvalScoreIn(BaseModel):
    interview_id: str | None = None
    repeats: int = 5
